go/vt/topo/srv_keyspace.go

```diff
@@ -115,6 +124,403 @@ func (ts *Server) GetSrvKeyspaceNames(ctx context.Context, cell string) ([]strin
 	}
 }
 
+// GetShardServingCells returns cells where this shard is serving
+func (ts *Server) GetShardServingCells(ctx context.Context, si *ShardInfo) (servingCells []string, err error) {
```

Comment:
This makes sense. I'll update accordingly.

Reply:
Sugu - I was looking into this a bit more and I was thinking that it might be better do this in a future iteration. 

There are some parts where I will need to make some slight modifications. For instance: 
```
	servingCells, err := wr.ts.GetShardServingCells(ctx, shardInfo)
	if err != nil {
		return err
	}
	// Check the Serving map for the shard, we don't want to
	// remove a serving shard if not absolutely sure.
	if !evenIfServing && len(servingCells) > 0 {
		return fmt.Errorf("shard %v/%v is still serving, cannot delete it, use even_if_serving flag if needed", keyspace, shard)
	}
```
or
```
func (wr *Wrangler) RemoveShardCell(ctx context.Context, keyspace, shard, cell string, force, recursive bool) error {
	shardInfo, err := wr.ts.GetShard(ctx, keyspace, shard)
	if err != nil {
		return err
	}

	shardServingCells, err := wr.ts.GetShardServingCells(ctx, shardInfo)

	// check the cell is in the list already
	if !topo.InCellList(cell, shardServingCells) {
		return fmt.Errorf("cell %v in not in shard info", cell)
	}
```

It's not really big, but given that this PR is already huge. Thinking that making this change as a separate PR makes sense.